File: src/main/java/com/cjw/reactivecommunityproject/common/security/service/JwtService.java
package com.cjw.reactivecommunityproject.common.security.service;

import com.cjw.reactivecommunityproject.common.security.model.SecurityAccessJwt;

public interface JwtService {
    String createAccessToken(SecurityAccessJwt securityAccessJwt);

    String createRefreshToken(String userUid);

    Boolean validateToken(String token);

    SecurityAccessJwt getClaims(String token);
}

File: src/main/kotlin/com/devcjw/reactivecommunity/auth/service/JwtService.kt
package com.devcjw.reactivecommunity.auth.service

import com.devcjw.reactivecommunity.auth.model.RcUserJwtClaims

interface JwtService {
    fun createAccessToken(uid: String, level: Long): String

    // TODO: Refresh Token 은 payload 가 필요 없다
    fun createRefreshToken(uid: String, level: Long): String

    fun validateToken(token: String): Boolean

    fun getRcUser(token: String): RcUserJwtClaims
}